core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt

```diff
@@ -5,22 +5,24 @@ package software.aws.toolkits.core.credentials
 
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
 
-abstract class ToolkitCredentialsIdentifier {
+interface CredentialIdentifier {
     /**
-     * The ID must be unique across all [ToolkitCredentialsIdentifier].
-     * It is recommended to concatenate the factory ID into this field to help enforce this requirement.
+     * A user friendly display name shown in the UI.
      */
-    abstract val id: String
+    val displayName: String
 
     /**
-     * A user friendly display name shown in the UI.
+     * An optional shortened version of the name to display in the UI where space is at a premium
      */
-    abstract val displayName: String
+    val shortName: String get() = displayName
+}
 
+abstract class ToolkitCredentialsIdentifier : CredentialIdentifier {
```

Comment:
Maybe rename `CredentialIdentifierBase` ? I'm kinda following the same pattern with the `ProjectAccountSettingsManager` refactor in that I'm thinking:

```kt

interface AwsConnectionManager

abstract class AwsConnectionManagerBase : AwsConnectionManager

class DefaultAwsConnectionManager(delegate: AwsConnectionManagerBase) : AwsConnectionManager by delegate
```